diff --git a/lib/authorizer.js b/lib/authorizer.js
index 7a4c979..a781b2f 100644
--- a/lib/authorizer.js
+++ b/lib/authorizer.js
@@ -24,7 +24,7 @@ module.exports.authorizePublish = function (client, topic, payload, callback) {
     let flag = false;
     let tops = topic.split('/');
 
-    if (client.super_user || client.id == topic.split('/')[0]) {
+    if (client.super_user || client.id == tops[0]) {
         flag = true;
     }
     else if (tops.length >= 3 && "kp" == tops[0] && client.id == tops[2]) {
@@ -39,8 +39,15 @@ module.exports.authorizePublish = function (client, topic, payload, callback) {
 
 module.exports.authorizeSubscribe = function (client, topic, callback) {
     let flag = false;
+    let tops = topic.split('/');
 
-    if (client.super_user || client.id == topic.split('/')[0]) {
+    if (client.super_user || client.id == tops[0]) {
+        flag = true;
+    }
+    else if (tops.length >= 3 && "kp" == tops[0] && client.id == tops[1]) {
+        flag = true;
+    }
+    else if (tops.length >= 3 && "kp" == tops[0] && "FFFFFFFFFFFF" == tops[1]) {
         flag = true;
     }
 
